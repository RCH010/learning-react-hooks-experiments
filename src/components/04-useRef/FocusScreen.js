import React, { useRef } from 'react'
import '../02-useEffect/SimpleForm.css'

export const FocusScreen = () => {
    //este no es un caso normal, pero pues ejemplifica
    const inputRef = useRef()

    const handleClick = () => {
        // document.querySelector('#nombre').select();
        inputRef.current.select();
    }
    return (
        <div>
            <h1 className='h1'>Focus Screen</h1>
            <br/>
            <input
                ref = {inputRef}
                id='nombre'
                className='form-control'
                placeholder='Su nombre'
                type='text'
            />
            <br/>
            <button 
                className='btn btn-outline-primary'
                onClick={handleClick}
            >Focus</button>
        </div>
    )
}
