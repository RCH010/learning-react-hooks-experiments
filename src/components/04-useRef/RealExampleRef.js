import React, { useState } from 'react'
import '../02-useEffect/SimpleForm.css'
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks'

export const RealExampleRef = () => {
    
    const [show, setShow] = useState(false)

    return (
        <div>
            <h1 className='h1'>Real Example useRef</h1>
            <br />
            {show&& <MultipleCustomHooks />}
            <button 
                className='btn btn-primary'
                onClick={() => {setShow(!show)}}
            >Show/Hide</button>
        </div>
    )
}
