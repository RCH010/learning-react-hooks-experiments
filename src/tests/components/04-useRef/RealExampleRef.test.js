import { shallow } from "enzyme"
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef"

describe('Tests in UseRef realExample', () => {
    
    const wrapper = shallow(<RealExampleRef />);

    test('should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    })

    test('should display component <MultipleCustomHooks/>', () => {
        const btn = wrapper.find('button');
        btn.simulate('click');
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    })
})
