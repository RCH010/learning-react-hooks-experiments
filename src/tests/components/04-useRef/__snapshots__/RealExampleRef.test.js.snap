// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in UseRef realExample should display component <MultipleCustomHooks/> 1`] = `
<div>
  <h1
    className="h1"
  >
    Real Example useRef
  </h1>
  <br />
  <MultipleCustomHooks />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;

exports[`Tests in UseRef realExample should display correctly 1`] = `
<div>
  <h1
    className="h1"
  >
    Real Example useRef
  </h1>
  <br />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
